### DISCUSSION TITLE
Is there a way to make babel embed @babel/runtime?

### DISCUSSION THREAD

**Discussion by marco-a** [babel/babel#11611] [General] [answered]
I'm using `@babel/plugin-transform-runtime` to transpile `async/await` statements with babel.

Only problem I have is that `@babel/plugin-transform-runtime` needs `@babel/runtime` at runtime.

I just cannot find a simple way to get babel to inline the helper functions.

	var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

	var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

	var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

What I want:

	var _interopRequireDefault = /* resolved definition */

	var _regenerator = /* resolved definition */

	var _asyncToGenerator2 = /* resolved definition */

**Comment by nicolo-ribaudo** [comment, marked as answer]
You can try passing `helpers: false` to Babel runtime. However, `regenerator-runtime` would still use a `require()` statement because it's a separate package.

If you can't have `require()`s in your code, I suggest completely removing the `transform-runtime` plugin and manually adding `<script src="https://cdn.jsdelivr.net/npm/regenerator-runtime@0.13.5/runtime.js"></script>` to your HTML code (before loading your javascript files).

**Reply by marco-a** [reply]
Thank you for the swift response :)

Setting `helpers: false` did indeed remove most `require()` calls to `@babel/runtime`, great!

However I'm not using HTML.

Rather I want to deploy a npm package that is independent of `@babel/runtime`.

As said, I'd like to embed the whole (or needed) babel runtime in the transpiled code.

**Reply by nicolo-ribaudo** [reply]
Currently regenerator doesn't provide a way to inline its helpers, so you should bundle your code before publishing, so that it contains regenerator-runtime's code.

However, keep in mind that `regenerator-runtime` has 23M downloads/week (https://www.npmjs.com/package/regenerator-runtime), so even if you add it as a dependency most likely it's already in the `node_modules` folder of your library's users.

**Reply by marco-a** [reply]
Yes that's what I'm doing right now.

Would be cool if `@babel\plugin-transform-runtime` had an option `embed` (or something like that) that when set to `true` will embed all code.

This way, one doesn't have to bundle the transformed code :)
